TypeCobol/Tools/Options-Config/TypeCobolConfiguration.cs

```diff
@@ -368,12 +389,50 @@ public static OptionSet GetCommonTypeCobolOptions(TypeCobolConfiguration typeCob
             if (!CanCreateFile(config.LogFile) && !string.IsNullOrEmpty(config.LogFile))
                 errorStack.Add(ReturnCode.LogFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.LogFileError]);
 
+#if EUROINFO_RULES
+            //CPY Map File
+            try
+            {
+                config.CpyCopyNamesMap = GetCpyCopiesFile(config.CPYMapFilePath);
+            } catch(Exception e) {
+                //Fail to read the Copy File Name, Log
+                System.IO.File.AppendAllText(config.LogFile != null ? config.LogFile : TypeCobolConfiguration.DefaultLogFileName, e.ToString());
+                if (config.CPYMapFilePath != null)
+                {
+                    errorStack.Add(ReturnCode.CopyNameMapFileError, TypeCobolConfiguration.ErrorMessages[ReturnCode.CopyNameMapFileError]);
+                }
+            }
+#endif
             //CustomAnalyzers
             VerifFiles(config.CustomAnalyzerFiles, ReturnCode.CustomAnalyzerFileError, errorStack);
 
             return errorStack;
         }
 
+#if EUROINFO_RULES
+        /// <summary>
+        /// Get the file of CPY COPY names to be used.
+        /// </summary>
+        /// <param name="cpyCopiesFilePath"></param>
+        /// <exception cref="Exception">Any exception if an error occurs.</exception>
+        /// <returns>the CopyNameMapFile instance if one is available, null otherwise</returns>
+        public static CopyNameMapFile GetCpyCopiesFile(string cpyCopiesFilePath)
+        {
+            string cpyMapFilePath = cpyCopiesFilePath;
+            if (cpyMapFilePath == null)
+            {
+                cpyMapFilePath = Path.Combine(Path.GetDirectoryName(System.Reflection.Assembly.GetEntryAssembly().Location),
+                    TypeCobolConfiguration.DefaultCopyNameFile);
+                if (!File.Exists(cpyMapFilePath))
+                    cpyMapFilePath = null;
+            }
+            if (cpyMapFilePath != null)
+            {
+                return new CopyNameMapFile(cpyMapFilePath);                    
+            }
+            return null;
+        }
```

Comment:
Would go in TypeCobolOptions (setter of `CpyCopyNamesMapFilePath` as part of the load process). Also don't forget to check if the path is rooted (`Path.IsPathRooted`) to allow full path and relative path to be used.

Reply:
doen in e4234a0